### PR TITLE
Enabled accept errors in net-sim testing

### LINKED ISSUE #3315: prop_multinode_Sim: in presence of accept IO errors
This should reproduce the https://github.com/input-output-hk/cardano-node/issue/3080 problem in simulation.

### PR DESCRIPTION
**Pull request by coot**
Fixes #3315

- ouroboros-network-framework: added todo
- sim-net: removed IOErrThrowOrReturn as an abstraction leak
- sim-net: reset channel when accept errors
- net-sim: added abiAcceptFailure
- net-sim: refactor TestError
- net-sim: fix prop_connect_to_accepting_socket
- Fixed some typos

### CODE REVIEW

File: ouroboros-network-framework/src/Simulation/Network/Snocket.hs [hunk 1]
```diff
@@ -359,7 +350,7 @@ instance (Typeable addr, Show addr)
 
 
 -- | A type class for global IP address scheme.  Every node in the simulation
--- has an ephemeral address.  Every node in the simulation has an implicity ipv4
+-- has an ephemeral address.  Every node in the simulation has an implicite ipv4
```
Comment (bolt12) [inline ouroboros-network-framework/src/Simulation/Network/Snocket.hs:353]:
```diff
- -- has an ephemeral address.  Every node in the simulation has an implicite ipv4
+ -- has an ephemeral address.  Every node in the simulation has an implicit ipv4
```

File: ouroboros-network-framework/test/Test/Simulation/Network/Snocket.hs [hunk 2]
```diff
@@ -351,13 +351,14 @@ toBearerInfo abi =
 -- Properties
 --
 
-data TestError addr = UnexpectedOutcome String
+data TestError addr = UnexpectedOutcome
+                    | UnexpectedError SomeException
```
Comment (bolt12) [inline ouroboros-network-framework/test/Test/Simulation/Network/Snocket.hs:358]: 😄

**Review by bolt12 — APPROVED** [review]
LGTM!

**Comment by coot** [comment]
@bolt12 I removed the commit 8c6ed3d9c, I realised that it is not only harmful but also not needed at all.

**Comment by coot** [comment]
bors merge

**Pull request merged**